### ISSUE TITLE
feature request: changing delays while in autoplay

### ISSUE THREAD

**Issue by davidobber** [serenader2014/flutter_carousel_slider#90]
Add the ability to change the speed while in autoplay. Now it's possible only when it's paused because getTimer() is called when the carousel starts after pause and in initState()

**Comment by serenader2014** [comment]
Hi, sorry for the late response.

Currently the plugin support `autoPlayAnimationDuration` option which allow you to control the sliding animation duration, is that what you need?

**Comment by davidobber** [comment]
I'm sorry but I'm focused to something else and in the last two months I didn't write any mobile code. I don't remember exactly the topic but if I remember well I tried to slow down the animation with a timer. I tried to use autoPlayAnimationDuration without pausing the animation. But I was not able to do it.

**Comment by Solesty** [comment]
I don't think it is possible with the current version of the plugin, but I got it working by using a timer.

` Timer timer;
  _reinitiateTimer() {
    timer?.cancel();
    timer = Timer.periodic(Duration(seconds: this._playInterval), (Timer t) {
      if (carouselController.ready) {
        carouselController.nextPage(
            curve: Curves.easeInOut, duration: Duration(seconds: 1));
      }
    });
  }`

I change the _playInterval variable in a setState. You have to call the _reinitiateTimer method inside your build before building any thing. This is a workaround. 

Don't forget to cancel the timer in your dispose method as well.

**Comment by davidobber** [comment]
Thank you!

David

Il giorno ven 17 apr 2020 alle ore 09:54 Aboyeji, Adeleke Solomon <
notifications@github.com> ha scritto:
